ext/geos_c_impl/geometry_collection.c

```diff
@@ -66,9 +66,14 @@ create_geometry_collection(VALUE module, int type, VALUE factory, VALUE array)
   for (i = 0; i < len; ++i) {
     geom = rgeo_convert_to_detached_geos_geometry(
       rb_ary_entry(array, i), factory, cast_type, &klass, &state);
-    if (state || !geom) {
-      break;
+    if (state) {
+      for (j = 0; j < i; j++) {
+        GEOSGeom_destroy(geoms[j]);
+      }
+      FREE(geoms);
+      rb_jump_tag(state);
```

Comment:
nice! Maybe we could remove all of my `rb_exc_raise(rb_errinfo())` with that, seems like the way to go!

Reply:
I didn't change the ones that were already in there, but I will change to that now.